bin/varnishd/mgt/mgt_vcl.c

```diff
@@ -594,16 +606,60 @@ static void v_matchproto_(cli_func_t)
 mcf_vcl_load(struct cli *cli, const char * const *av, void *priv)
 {
 	struct vclprog *vp;
+	const char *vcls[] = {av[3], NULL};
 
 	(void)priv;
 	if (!mcf_find_no_vcl(cli, av[2]))
 		return;
 
-	vp = mgt_new_vcl(cli, av[2], NULL, av[3], av[4], 0);
+	vp = mgt_new_vcl(cli, av[2], NULL, vcls, av[4], NULL, 0);
 	if (vp != NULL && !MCH_Running())
 		VCLI_Out(cli, "VCL compiled.\n");
 }
 
+static void v_matchproto_(cli_func_t)
+mcf_vcl_load_files(struct cli *cli, const char * const *av, void *priv)
+{
+	struct vclprog *vp;
+	const struct parspec *pp;
+	int i = 2;
+	const char *state = NULL;
+	const char *vcl_path = NULL;
+
+	while (av[i] != NULL) {
+		if (!strcmp(av[i], "-s")) {
+			FAIL_NO_ARG(av[i+1], "-s");
+			state = av[i+1];
+			i += 2;
+		} else if (!strcmp(av[i], "-p")) {
+			FAIL_NO_ARG(av[i+1], "-p");
+			pp = MCF_FindPar("vcl_path");
+			if (pp->flags & PROTECTED) {
+				VCLI_SetResult(cli, CLIS_AUTH);
+				VCLI_Out(cli, "parameter \"vcl_path\" is protected.\n");
+				return;
+			}
```

Comment:
To clarify my previous comment, since both `varnishd -r` and `vcl.load_files` require admin privileges as per our security model, I would be fine relaxing this rule and allowing both a read-only global `vcl_path` and per-VCL additional path since they take precedence.